src/control/common/system.go

```diff
@@ -91,60 +133,49 @@ func (smrs SystemMemberResults) String() string {
 type Membership struct {
 	sync.RWMutex
 	log     logging.Logger
-	members map[string]SystemMember
+	members SystemMembers
 }
 
-// Add adds member to membership.
-func (m *Membership) Add(member SystemMember) (int, error) {
+// Add adds member to membership, returns member count.
+func (m *Membership) Add(member *SystemMember) (int, error) {
 	m.Lock()
 	defer m.Unlock()
 
-	if value, found := m.members[member.Uuid]; found {
-		return -1, errors.Errorf("member %s already exists", value)
-	}
-
-	m.members[member.Uuid] = member
-
-	return len(m.members), nil
+	return m.members.Add(member)
 }
 
-// Remove removes member from membership, idenpotent.
-//
-// Avoid taking a RW lock where possible.
-func (m *Membership) Remove(uuid string) {
+// Remove removes member from membership, idempotent.
+func (m *Membership) Remove(rank uint32) {
 	m.Lock()
 	defer m.Unlock()
 
-	delete(m.members, uuid)
+	delete(m.members, rank)
 }
 
 // GetMember retrieves member from membership based on UUID.
-func (m *Membership) GetMember(uuid string) (*SystemMember, error) {
+func (m *Membership) GetMember(rank uint32) (*SystemMember, error) {
 	m.RLock()
 	defer m.RUnlock()
 
-	member, found := m.members[uuid]
-	if !found {
-		return nil, errors.Errorf("member with uuid %s not found", uuid)
-	}
-
-	return &member, nil
+	return m.members.GetMember(rank)
 }
 
-// GetMembers returns internal member structs as a sequence.
+// GetMembers returns SystemMembers.
```

Comment:
This change removes all thread safety. Not sure if that was intended, but now callers of this method will have free access to the previously-protected private membership map.